packages/suite/src/components/wallet/TransactionItem/CoinjoinBatchItem.tsx

```diff
@@ -163,10 +168,21 @@ export const CoinjoinBatchItem = ({
     isPending,
 }: CoinjoinBatchItemProps) => {
     const lastTx = transactions[0];
-    const amount = sumTransactions(transactions);
-    const fiatAmount = sumTransactionsFiat(transactions, localCurrency);
     const { FiatAmountFormatter } = useFormatters();
-    const isMissingFiatRates = transactions.some(tx => !tx.rates?.[localCurrency]);
+    const historicFiatRates = useSelector(selectHistoricFiatRates);
+    const amount = sumTransactions(transactions);
+    const fiatAmount = sumTransactionsFiat(transactions, localCurrency, historicFiatRates);
+    const isMissingFiatRates = transactions.some(tx => {
+        const fiatRateKey = getFiatRateKey(
+            tx.symbol,
+            localCurrency as FiatCurrencyCode,
```

Comment:
unnecesary cast, the prop of component can be changed to `FiatCurrencyCode` type